gcs_client.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef __GCS_CLIENT_HPP_
```

Comment:
nit: global names starting with two underscores are reserved for the implementation.

If you don't need to follow the Google Style Guide, consider `#pragma once`.